test/node-http-test.js

```diff
@@ -0,0 +1,285 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: 'off' */
+
+'use strict';
+
+const assert = require('bsert');
+const Network = require('../lib/protocol/network');
+const FullNode = require('../lib/node/fullnode');
+const rules = require('../lib/covenants/rules');
+const common = require('./util/common');
+const network = Network.get('regtest');
+
+const { NodeClient, WalletClient } = require('hs-client');
+
+const nclient = new NodeClient({
+  port: network.rpcPort,
+  apiKey: 'foo'
+});
+
+const wclient = new WalletClient({
+  port: network.walletPort,
+  apiKey: 'foo'
+});
+
+const wallet = wclient.wallet('primary');
+
+describe('Node http', function() {
+  this.timeout(10000);
+  const witnessedBlocks = {};
+  let NAME0, NAME1;
+  let node, mineBlocks, cbAddress;
+
+  beforeEach(async () => {
+    assert.equal(network.names.auctionStart, 0);
+
+    node = new FullNode({
+      memory: true,
+      apiKey: 'foo',
+      network: 'regtest',
+      workers: true,
+      plugins: [require('../lib/wallet/plugin')]
+    });
+
+    node.on('connect', (entry, block) => {
+      const blockHash = block.hash().toString('hex');
+      witnessedBlocks[blockHash] = blockHash;
+    });
+
+    mineBlocks = common.constructBlockMiner(node, nclient);
+    NAME0 = await rules.grindName(10, 0, network);
+    NAME1 = await rules.grindName(10, 20, network);
+
+    await node.open();
+
+    cbAddress = (await wallet.createAddress('default')).address;
+
+    await mineBlocks(network.names.auctionStart, cbAddress);
+  });
+
+  afterEach(async () => {
+    await node.close();
+  });
+
+  describe('getNameInfo', () => {
+    describe('For names that are available at height 0', () => {
+      it('It should return null when there hasn\'t been an auction initiated', async () => {
+        const nameInfo = await nclient.get(`/name/${NAME0}`);
+        assert.deepEqual(nameInfo, {
+          info: null,
+            start: {
+              reserved: false,
+              start: 0,
+              week: 0
+            }
+        });
+      });
+      it('It should start an auction on the first day', async () => {
+        const nameInfo = await nclient.get(`/name/${NAME0}`);
+        assert.deepEqual(nameInfo, {
+          info: null,
+          start: {
+            reserved: false,
+            start: 0,
+            week: 0
+          }
+        });
+
+        await mineBlocks(10, cbAddress);
+
+        const open = await wclient.execute('sendopen', [NAME0]);
+        assert(open);
+      });
+      it('It should start an auction on the 2nd day', async () => {
+        await mineBlocks(40, cbAddress);
```

Comment:
`40` is a magic number here, it is not immediately obvious how it is tied to the 2nd day